money-core/src/main/scala/com/comcast/money/core/Formatters.scala

```diff
@@ -33,14 +48,25 @@ object Formatters {
     new SpanId(traceId, parentId.toLong, selfId.toLong)
   }
 
-  def toHttpHeader(spanId: SpanId): String =
-    HttpHeaderFormat.format(spanId.traceId, spanId.parentId, spanId.selfId)
+  def toHttpHeader(spanId: SpanId): String = HttpHeaderFormat.format(spanId.traceId, spanId.parentId, spanId.selfId)
 
   def fromB3HttpHeaders(traceId: String, maybeParentSpanId: Option[String], maybeSpanId: Option[String]) = Try {
     (maybeParentSpanId, maybeSpanId) match {
-      case (Some(ps), Some(s)) => new SpanId(traceId, ps.toLong, s.toLong)
-      case (Some(ps), _) => new SpanId(traceId, ps.toLong)
-      case _ => new SpanId(traceId)
+      case (Some(ps), Some(s)) => new SpanId(traceId.toGuid, ps.fromHexStringToLong, s.fromHexStringToLong)
+      case (Some(ps), _) => new SpanId(traceId.toGuid, ps.fromHexStringToLong)
+      case (_, Some(s)) => new SpanId(traceId.toGuid, s.fromHexStringToLong, s.fromHexStringToLong) // root span
```

Comment:
rather than comment what do you think about just making these functions inside of fromB3Headers?